reference-implementation/lib/writable-stream.js

```diff
@@ -919,6 +929,9 @@ function WritableStreamDefaultControllerProcessWrite(controller, chunk) {
       WritableStreamDefaultControllerAdvanceQueueIfNeeded(controller);
     },
     reason => {
+      if (stream._state === 'writable') {
```

Comment:
This check does not appear [in the spec text](https://streams.spec.whatwg.org/commit-snapshots/323223e65455e1f22be656b01fe2d810902db2ac/#writable-stream-default-controller-process-write). I believe it should be removed from the reference implementation?

Reply:
Great catch! Actually this check is required, and I failed to add it to the spec text. Without this check, if a write() rejects while an abort() is already in progress, the abortAlgorithm will be cleared too early and so when the time comes to call the underlying source abort() it will fail.